packages/fuel-indexer-database/database-types/src/lib.rs

```diff
@@ -446,3 +447,90 @@ impl IdCol {
         "ID".to_string()
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum QueryElement {
+    Field { key: String, value: String },
+    ObjectOpeningBoundary { key: String },
+    ObjectClosingBoundary,
+}
+
+// TODO: Adjust filter to allow for more complex filtering
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct QueryFilter {
+    pub key: String,
+    pub relation: String,
+    pub value: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct UserQuery {
+    pub elements: Vec<QueryElement>,
+    pub joins: Vec<String>,
+    pub namespace_identifier: String,
+    pub entity_name: String,
+    pub filters: Vec<QueryFilter>,
+}
+
+impl UserQuery {
+    pub fn to_sql(&self, db_type: &DbType) -> String {
+        match db_type {
+            DbType::Postgres => {
+                let mut peekable_elements = self.elements.iter().peekable();
+
+                let mut elements = Vec::new();
```

Comment:
Lots of business logic here and by reading the code it's not completely clear what's going on. Maybe some comments?